tenacity/__init__.py

```diff
@@ -287,6 +295,14 @@ def statistics(self) -> t.Dict[str, t.Any]:
             self._local.statistics = t.cast(t.Dict[str, t.Any], {})
             return self._local.statistics
 
+    @property
+    def iter_state(self) -> IterState:
+        try:
+            return self._local.iter_state  # type: ignore[no-any-return]
+        except AttributeError:
+            self._local.iter_state = t.cast(IterState, {})
```

Comment:
…and avoid a cast here?

Reply:
https://github.com/jd/tenacity/pull/434/commits/dfd09074e62684e376264a5eb548645759797edd